networkit/cpp/io/MemoryMappedFile.cpp

```diff
@@ -6,6 +6,24 @@ namespace NetworKit {
 	MemoryMappedFile::MemoryMappedFile(const std::string &path) { open(path); }
 
 	MemoryMappedFile::~MemoryMappedFile() { close(); }
+
+    MemoryMappedFile::MemoryMappedFile(MemoryMappedFile&& o) noexcept {
+		*this = std::move(o);
+	}
+
+    MemoryMappedFile& MemoryMappedFile::operator=(MemoryMappedFile&& o) noexcept {
+	    close();
```

Comment:
This fails if someone this `x = std::move(x);` for a `MemoryMappedFile` `x`. Move assignment should almost always be implemented using `swap` to avoid such problems. Other than that, the PR is fine.

Reply:
I fixed it by adding a check into the move-assignment operator. This is a no-brainer, since we are not talking about hot-code. Nevertheless I would argue that this is not an issue:
 - either a user move a unique temporary, then this cannot happen
 - if one writes `x = std::move(x)` he's clearly saying on the RHS that `x` is allows to be invalidated

AFAIK for this reason STL forbids these constructs in [res.on.arguments].